app/helpers/Content/Guide.php

```diff
@@ -207,4 +217,38 @@ public function getSEOTitle()
             return $this->getTitle();
         }
     }
+
+    /**
+     * @return string
+     */
+    public static function fixDuplicateAnchorIDs($content, $anchors)
+    {
+        $anchorList = [];
+        foreach ($anchors as $anchor) {
+            $anchorList[$anchor['subsectionId']][] = $anchor['sectionId'];
+        }
+        $dom = new \DomDocument();
+        // Will probably be duplicate ID warnings so suppress them
+        @$dom->loadHtml($content);
+        $anchors = $dom->getElementsByTagName('*');
+        $ids=[];
+        $duplicateId = 0;
+        foreach ($anchors as $anchor) {
+            $id = $anchor->getAttribute('id');
+            if ($id && isset($anchorList[$id]) && in_array($id, $ids)) {
+                // should only amend the ID via $anchorList if it is a h3
+                if ($anchor->tagName == 'h3') {
+                    $parent = array_shift($anchorList[$id]);
+                    $anchor->setAttribute('id', "$parent-$id");
+                } else {
+                    // if it is not a h3 amend so the ID is not duplicated
+                    $duplicateId++;
+                    $anchor->setAttribute('id', "$id-$duplicateId");
+                }
+            }
+            $ids[] = $id;
+        }
```

Comment:
a suggestion to not store the parent in an array

```suggestion
        foreach ($anchors as $anchor) {
            $anchorList[$anchor['subsectionId']] = $anchor['sectionId'];
        }
        $dom = new \DomDocument();
        // Will probably be duplicate ID warnings so suppress them
        @$dom->loadHtml($content);
        $anchors = $dom->getElementsByTagName('*');
        $ids = [];
        $duplicateId = 0;
        foreach ($anchors as $anchor) {
            $id = $anchor->getAttribute('id');
            if ($id && isset($anchorList[$id]) && in_array($id, $ids)) {
                // should only amend the ID via $anchorList if it is a h3
                if ($anchor->tagName == 'h3') {
                    $parentId = $anchorList[$id];
                    $anchor->setAttribute('id', "$parentId-$id");
                } else {
                    // if it is not a h3, amend so the ID is not duplicated
                    $duplicateId++;
                    $anchor->setAttribute('id', "$id-$duplicateId");
                }
            }
            $ids[] = $id;
        }
```

Reply:
Tested it and it does not work the anchors end up all getting the same id

Reply:
All good, thanks for giving it a go. I don't have an instance up to debug this further, so let's leave it as is.